tests/integration/support/steps.go

```diff
@@ -343,6 +365,67 @@ func (c *ScenarioCtx) PayloadsOfDispatchedEventsDontIncludeDecisions() error {
 	return nil
 }
 
+// TheUserProfileServiceStateShouldBe checks current state of UPS
+func (c *ScenarioCtx) TheUserProfileServiceStateShouldBe(value *gherkin.DocString) error {
+	config, err := c.clientWrapper.Client.GetProjectConfig()
+	if err != nil {
+		return fmt.Errorf("Invalid Project Config")
+	}
+	parsedProfiles, err := parseYamlArray(value.Content, config)
+	if err != nil {
+		return fmt.Errorf("Invalid request for user profile service state")
+	}
+
+	var expectedProfiles []decision.UserProfile
+	for _, profile := range parsedProfiles {
```

Comment:
This can be another util method inside of the `optlyplugins/user_profile_service` package